src/components/UrlUtilsService.js

```diff
@@ -48,13 +48,44 @@ goog.provide('ga_urlutils_service');
                   /.*kmz$/.test(url));
         };
 
-        this.proxifyUrl = function(url) {
+        // Test using a head request if the remote resource enables CORS
+        this.isCorsEnabled = function(url) {
+          var deferred = $q.defer();
+          $http.head(url, { timeout: 1500 }).then(function(response) {
+            // If the head request responded successfully it means that CORS
+            // is enabled
```

Comment:
yes using curl and being in the browser is 2 different things. The browser will block the resource and as a result the request will be blocked.
I'll provide more test cases.